src/audio/src/src.c

```diff
@@ -208,6 +208,12 @@ static int src_buffer_lengths(struct comp_dev *dev, struct comp_data *cd,
 		 */
 		a->sbuf_length = 2 * nch * stage1->blk_out * r1;
 		a->sbuf_length += a->sbuf_length >> 3;
+
+		/* sbuf_length is used to compute buffer_start which needs to
+		 * be aligned to 8 bytes as required by some Xtensa
+		 * instructions (e.g AE_L32X2F24_XC)
+		 */
+		a->sbuf_length = ALIGN_UP(a->sbuf_length, 8);
```

Comment:
Sorry I overlooked this in review and noticed my previous patch made a big mistake. The sbuf_length is not size in bytes but length as samples. The value must be an integer multiple of channel count. Making it multiple of 8 would not work for e.g. 6 channels.

So something like

```
n = 2 * stage1->blk_out * r1;
a->sbuf_length = nch * (n + (n >> 3));
```

@dbaluta Could you please try if above patch works for the bug you found.

Reply:
Still one thing missing. If channels count is odd, then buffer_start would not be 8 bytes aligned. So there's need in src_params_general() for

`delay_lines_size = ALIGN_UP(sizeof(int32_t) * cd->param.total, 8);
`
and get bugger_start to same 8 bytes align

`buffer_start = ALIGN_UP(cd->delay_lines + cd->param.sbuf_length, 8);
`

Reply:
The last macro usage does not seem to work, maybe this instead

`buffer_start = cd->delay_lines + ALIGN_UP(cd->param.sbuf_length, 2); `

Reply:
@singalsu following change based on your suggestion works for me:

```
--- a/src/audio/src/src.c
+++ b/src/audio/src/src.c
@@ -141,7 +141,7 @@ static int src_buffer_lengths(struct comp_dev *dev, struct comp_data *cd,
        struct src_param *a;
        int fs_in, fs_out;
        int source_frames;
-       int r1;
+       int r1, n;
 
        a = &cd->param;
        fs_in = cd->source_rate;
@@ -206,8 +206,8 @@ static int src_buffer_lengths(struct comp_dev *dev, struct comp_data *cd,
                 * variable number of blocks to process per each stage
                 * there is no equation known for minimum size.
                 */
-               a->sbuf_length = 2 * nch * stage1->blk_out * r1;
-               a->sbuf_length += a->sbuf_length >> 3;
+               n = 2 * stage1->blk_out * r1;
+               a->sbuf_length = nch * (n + (n >> 3));
        }
 
        a->src_multich = a->fir_s1 + a->fir_s2 + a->out_s1 + a->out_s2;
@@ -771,7 +771,7 @@ static int src_params_general(struct comp_dev *dev, struct comp_data *cd,
                goto out;
        }
 
-       delay_lines_size = sizeof(int32_t) * cd->param.total;
+       delay_lines_size = ALIGN_UP(sizeof(int32_t) * cd->param.total, 8);
        if (delay_lines_size == 0) {
                comp_err(dev, "src_params(): delay_lines_size = 0");
 
@@ -792,7 +792,7 @@ static int src_params_general(struct comp_dev *dev, struct comp_data *cd,
 
        /* Clear all delay lines here */
        memset(cd->delay_lines, 0, delay_lines_size);
-       buffer_start = cd->delay_lines + cd->param.sbuf_length;
+       buffer_start = cd->delay_lines + ALIGN_UP(cd->param.sbuf_length, 2);
 
        /* Initialize SRC for actual sample rate */
        n = src_polyphase_init(&cd->src, &cd->param, buffer_start);

```

Anyhow, I'm not sure why this is entirely correct.

We need bufer_start to be aligned to 8 but we compute it as:

```
buffer_start = cd->delay_lines + ALIGN_UP(cd->param.sbuf_length, 2); 
```

Reply:
Also if doing that way, the param.total could be changed to param.total_size_aligned.

Reply:
@singalsu OK, now I understand the explanation. Will go with my patch above.